paasta_tools/contrib/paasta_update_soa_memcpu.py

```diff
@@ -165,29 +170,83 @@ def get_report_from_splunk(creds, app, filename, criteria_filter):
         criteria = d["result"]["criteria"]
-        serv = {}
-        serv["service"] = criteria.split(" ")[0]
-        serv["cluster"] = criteria.split(" ")[1]
-        serv["instance"] = criteria.split(" ")[2]
-        serv["owner"] = d["result"].get("service_owner", "Unavailable")
-        serv["date"] = d["result"]["_time"].split(" ")[0]
-        serv["money"] = d["result"].get("estimated_monthly_savings", 0)
-        serv["project"] = d["result"].get("project", "Unavailable")
-        serv["cpus"] = d["result"].get("suggested_cpus")
-        serv["old_cpus"] = d["result"].get("current_cpus")
-        serv["mem"] = d["result"].get("suggested_mem")
-        serv["old_mem"] = d["result"].get("current_mem")
-        serv["disk"] = d["result"].get("suggested_disk")
-        serv["old_disk"] = d["result"].get("current_disk")
-        serv["min_instances"] = d["result"].get("suggested_min_instances")
-        serv["max_instances"] = d["result"].get("suggested_max_instances")
-        serv["hacheck_cpus"] = d["result"].get("suggested_hacheck_cpus")
-        serv["cpu_burst_add"] = d["result"].get("suggested_cpu_burst_add")
-        services_to_update[criteria] = serv
+
+        # fill in fields that don't need any transformation
+        serv = {
+            "service": criteria.split(" ")[0],
+            "cluster": criteria.split(" ")[1],
+            "instance": criteria.split(" ")[2],
+            "owner": d["result"].get("service_owner", "Unavailable"),
+            "date": d["result"]["_time"].split(" ")[0],
+            "money": d["result"].get("estimated_monthly_savings", 0),
+            "project": d["result"].get("project", "Unavailable"),
+        }
+
+        # the report we get is all strings, so we need to convert them to the right types
+        field_conversions = {
+            # the general format here is:
+            # expected_key: (source_key, conversion_function)
+            # where expected_key is the key we want in the final output and source_key is the key from the splunk
+            # report
+            "cpu_burst_add": ("suggested_cpu_burst_add", float),
+            "cpus": ("suggested_cpus", float),
+            "disk": ("suggested_disk", int),
+            "hacheck_cpus": ("suggested_hacheck_cpus", float),
+            "max_instances": ("suggested_max_instances", int),
+            "mem": ("suggested_mem", int),
+            "min_instances": ("suggested_min_instances", int),
+            "old_cpus": ("current_cpus", float),
+            # not quite sure why these are floats...they're ints in soaconfigs
+            "old_disk": ("current_disk", _force_str_to_int),
+            "old_mem": ("current_mem", _force_str_to_int),
+        }
+
+        for key, (source_key, conversion_function) in field_conversions.items():
+            value = d["result"].get(source_key)
+            serv[key] = conversion_function(value) if value is not None else None
```

Comment:
I don't really know why we never had to convert these from strings before, but if we suspect this could be causing more autotune differences than we expect w/ this change, maybe we could just pass conversion_function as `max(v1, v2, key=conversion_function)` instead?

Reply:
:o - clever!

i think that we end up converting later on, but this does sound like a less invasive change

Reply:
(i did this)